source/Sailfish/Execution/SailfishExecutionEngine.cs

```diff
@@ -165,12 +165,9 @@ await mediator.Publish(
                     {
                         await testCase.CoreInvoker.GlobalSetup(cancellationToken);
```

Comment:
Consider throwing an exception or logging something when we expect executionState to have something which I think is when ShouldCallGlobalSetup returns false.

Reply:
ShouldCallGlobalSetup computes where in the set of test case instances for a given class the current iteration is. 

So if you think of a Sailfish class as a type for which we create many instances (test cases) which are then executed in order - then the method tells you if you're on the first instance or not. The global teardown method tells you if you're on the last test case for the class.

This code could be written more nicely.

If the global setup method itself throws (e.g. user error), then we catch and return.